test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs

```diff
@@ -321,5 +319,24 @@ public class ActionResultTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
             Assert.Equal("{\"SampleInt\":10,\"SampleString\":\"Foo\"}", await response.Content.ReadAsStringAsync());
         }
+
+        [Fact]
+        public async Task HttpNotFoundObjectResult_WithDisposableObject()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName);
+            var client = server.CreateClient();
+
+            // Act
+            var response1 = await client.GetAsync("/ActionResultsVerification/GetDisposeCallCount");
+            await client.GetAsync("/ActionResultsVerification/GetNotFoundObjectResultWithDisposableObject");
+            var response2 = await client.GetAsync("/ActionResultsVerification/GetDisposeCallCount");
+
+            // Assert
+            var initialCallCount = Convert.ToInt32(await response1.Content.ReadAsStringAsync());
+            var finalCallCount = Convert.ToInt32(await response2.Content.ReadAsStringAsync());
+            Assert.True(initialCallCount == 0);
```

Comment:
There are more specific asserts than these. `Assert.True` should generally be used only for asserting actual bool's.